Feature/sof 6727

VsevolodX opened the pull request:
Draft PR

VsevolodX committed:
feat: change toolbar to use MUI - initial commit

VsevolodX committed:
=initial draft

VsevolodX committed:
initial draft commit

Added TS, imported cove.js package and necessary packages for it to work, viewSettingsToolbar is moved to Dropdown and actions work,
probably broke that one when updating the cove.js locally

VsevolodX committed:
feature: add nested dropdown to cove.js to use in wave.js

VsevolodX committed:
feat: add nested dropdown

VsevolodX committed:
feat: add nested dropdown - 2

timurbazhirov commented on .eslintrc.json:
```diff
@@ -3,5 +3,34 @@
+            "plugins": ["@typescript-eslint"]
+        }
+    ]
+}
```
https://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file

timurbazhirov commented on package.json:
```diff
@@ -55,7 +56,10 @@
-        "underscore.string": "^3.3.4"
+        "underscore.string": "^3.3.4",
+        "@exabyte-io/cove.js": "2023.7.14-0",
+        "@mui/lab": "^5.0.0-alpha.120",
```
Why do we need this?

VsevolodX replied:
It was a solution to import cove.js, from error messages those libraries were missing. 

Without them:
![image](https://github.com/Exabyte-io/wave.js/assets/79542055/f67e0240-0a0d-4681-b94f-62536f5590fb)

They are peer dependencies in cove.js:
![image](https://github.com/Exabyte-io/wave.js/assets/79542055/7980a2a2-bf2a-432e-af32-eced52f23d30)

timurbazhirov replied:
@zoomchik - do we need this? Any way to avoid?

zoomchik replied:
@timurbazhirov We need this packages as peerDependencies at least to avoid multiple copies of React. https://legacy.reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react

timurbazhirov commented on src/components/IconToolbar.jsx:
```diff
@@ -27,14 +27,13 @@ export class IconToolbar extends React.Component {
         return (
             <div className={setClass(className, { hidden: isHidden })} data-name={title}>
-                <RoundIconButton
+                <SquareIconButton
```
What was the point of making this change? It distracts from important changes and doesn't introduce new functionality - dropdown is what's important in this PR

VsevolodX replied:
By default MUI IconButton is round, needs `border-radius=0` to change that to square, so it was the first attempt to create what's needed.

timurbazhirov commented on src/components/ThreeDEditor.jsx:
```diff
@@ -1,9 +1,15 @@
 
+// import Dropdown from "@exabyte-io/cove.js/dist/mui/components/dropdown/Dropdown";
+// import { DropdownItem } from "@exabyte-io/cove.js/dist/mui/components/dropdown/DropdownItem";
+// import NestedDropdown from "@exabyte-io/cove.js/dist/mui/components/nested-dropdown/NestedDropdown";
```
![dafuq](https://user-images.githubusercontent.com/721112/209405526-d56fc0a1-a610-4390-b75f-bb35534eed9a.gif)

timurbazhirov commented on src/components/ThreeDEditor.jsx:
```diff
@@ -33,10 +40,19 @@ import React from "react";
+// import ToolbarMenuItem from "./ToolbarMenu/ToolbarMenuItem.tsx";
 import { WaveComponent } from "./WaveComponent";
 
+const darkTheme = createTheme({
```
We should use the cove.js theme

timurbazhirov commented on src/components/dropdown/DefaultDropdownButton.tsx:
```diff
@@ -0,0 +1,56 @@
+    fullWidth: boolean;
+}
+
+export const DefaultDropdownButton = function ({
```
Why do we even need this component? It's just using the default Button

timurbazhirov commented on src/components/dropdown/Dropdown.tsx:
```diff
@@ -0,0 +1,154 @@
+/**
+ *  MUI dropdown component have a default button with dropdown also could be used with
+ * custom button which takes from children, actions array -> array which will be converted
+ * to dropdown menu items.
```
This sentence is not understandable, plus has many grammatic errors

VsevolodX replied:
Sure it is. I've noticed it too, but this was just a test so I just copied to use the component.

timurbazhirov commented on src/components/dropdown/Dropdown.tsx:
```diff
@@ -0,0 +1,154 @@
+                {...popperProps}
+            >
+                {({ TransitionProps }) => (
+                    <Grow {...TransitionProps} style={{ transformOrigin: "center top" }}>
```
Why do we need all this stuff?? There's a Dropdown component in cove.js that we should update to deal with the necessary logic

timurbazhirov commented on src/components/dropdown/NestedDropdown.tsx:
```diff
@@ -0,0 +1,154 @@
+                {...popperProps}
+            >
+                {({ TransitionProps }) => (
+                    <Grow {...TransitionProps} style={{ transformOrigin: "center top" }}>
```
Code here is exactly the same as in the component above - remove duplicates

timurbazhirov commented on src/react-app-env.d.ts:
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
WTH is this?

VsevolodX replied:
It was created during `npm run start`. I'll look into tsconfig.json, how it's handled in other packages.

VsevolodX committed:
feat: add nested dropdown - 3

VsevolodX committed:
feat: use button group for toolbar buttons (figma suggest this design))

VsevolodX committed:
update: convert actions into config and render via rendertoolbar()

zoomchik commented on src/components/ThreeDEditor.jsx:
```diff
@@ -22,14 +22,13 @@ import LooksIcon from "@mui/icons-material/Looks";
 import ThreeDRotation from "@mui/icons-material/ThreeDRotation";
-import { createTheme, ThemeProvider, Tooltip } from "@mui/material";
-// import { styled } from "@mui/system";;
+import { ButtonGroup, createTheme, ThemeProvider, Tooltip } from "@mui/material";
```
Do not import the whole lib, import individual components eg. `import from "@mui/material/Tooltip";`

VsevolodX committed:
update: cleanup

VsevolodX committed:
update: fix conventional cell checkmark toggling

VsevolodX committed:
update: fix linting errors

VsevolodX committed:
feat: render component for parameters and pass it to NestedDropdown

VsevolodX committed:
update: fix issues with linting

VsevolodX committed:
update: add dependency of cove.js SOF-6748-1

VsevolodX committed:
fix: fix jest tests not starting

VsevolodX committed:
update: change tests methods since rendering changed from css hide to conditional rendering

VsevolodX committed:
update: cleanup

VsevolodX committed:
fix: install git on centos to install intermediate changes from github

VsevolodX committed:
update: delete IconToolbar - not needed anywhere.

This component probably will be replaced with IconToolbar.tsx later on, when we isolate components.

VsevolodX committed:
update: commit package-lock for tests to run

VsevolodX committed:
update: move SquareIconButton.jsx to tsx

VsevolodX committed:
update: change linting rules to accommodate for TS

VsevolodX committed:
feat: create IconsToolbar.tsx component

VsevolodX committed:
feat: create ParametersMenu.tsx component and helper component for inout

VsevolodX committed:
update: use newly created components

VsevolodX committed:
update: remove unnecessary complexity in parameters menu component

VsevolodX committed:
update: add minor adjustments to resolve PR comments

VsevolodX committed:
update: use theme to style SquareIconButton

VsevolodX committed:
update: fix styling issues and use theme for that

VsevolodX committed:
Merge pull request #128 from Exabyte-io/SOF-6789

Sof-6789 - move render functions into components

VsevolodX committed:
update: setup Jest tests to handle TS

VsevolodX committed:
update: add key property to please the browser console

VsevolodX committed:
update: fix linting issue, use the same rule as in webapp

VsevolodX committed:
update: update package.json to use PR head of cove.js

VsevolodX committed:
remove unnecessary file

VsevolodX committed:
update: minor final fixes

VsevolodX committed:
remove unnecessary file

VsevolodX committed:
update: remove extensions of imported files

VsevolodX committed:
update: add a rule from web-app to solve import extension issue

VsevolodX committed:
update: remove unused CSS

VsevolodX committed:
update: minor adjustment

VsevolodX committed:
update: update cove.js import in package.json

VsevolodX committed:
update: extract groups of actions into getAction() methods for easy extension and override

VsevolodX committed:
update: update package.json to use latest cove.js

Commit:
chore: update package.json

Commit:
chore: update package.json

Commit:
chore: update package.json

VsevolodX committed:
update: remove dummy fullscreen button

VsevolodX committed:
Merge branch 'feature/SOF-6727' of github.com:Exabyte-io/wave.js into feature/SOF-6727

VsevolodX committed:
update: fix type issue

VsevolodX committed:
update: use cove.js pr

VsevolodX committed:
remove topmost dividers

VsevolodX committed:
update: update package-lock to be in sync

VsevolodX committed:
cleanup: minor fixes

VsevolodX committed:
update: add specific cove.js version

VsevolodX committed:
update: add paperPlacement prop

VsevolodX committed:
update: change import in package.json

VsevolodX committed:
fix: pass paperplacement property

VsevolodX committed:
update: use specific commit of cove.js

VsevolodX committed:
update: use latest cove.js

VsevolodX committed:
update: set specific theme.spacing

VsevolodX committed:
update: use new icon name

VsevolodX committed:
update: package.json to use latest cove.js

VsevolodX committed:
update: package-lock.json to use latest cove.js

VsevolodX committed:
update: add ts-ignore to address github warning

zoomchik commented on src/components/ParametersMenu.tsx:
```diff
@@ -68,6 +67,7 @@ function ParametersMenu(props: ParametersMenuProps) {
                                 id={`repetitionsAlongLatticeVector${label}`}
                                 value={
+                                    // @ts-ignore
                                     (viewerSettings as { [key: string]: any })[
```
Can we use ViewerSettings type above?

VsevolodX replied:
@zoomchik with help from Alex I went with defining type of key:
```
type ViewerSettingsKey = keyof ViewerSettings;
const key = `repetitionsAlongLatticeVector${label}` as ViewerSettingsKey;
```

VsevolodX committed:
fix: type issue
